hudi-aws/src/main/java/org/apache/hudi/aws/sync/AWSGlueCatalogClient.java

```diff
@@ -0,0 +1,483 @@
+          .withParameters(params)
+          .withPartitionKeys(schemaPartitionKeys)
+          .withStorageDescriptor(storageDescriptor)
+          .withLastAccessTime(new Date(System.currentTimeMillis()))
+          .withLastAnalyzedTime(new Date(System.currentTimeMillis()));
+      request.withDatabaseName(databaseName)
+          .withTableInput(tableInput);
+
+      CreateTableResult result = awsGlue.createTable(request);
+      LOG.info("Created table " + tableId(databaseName, tableName) + " : " + result);
+    } catch (AlreadyExistsException e) {
+      LOG.warn("Table " + tableId(databaseName, tableName) + " already exists.", e);
+    } catch (Exception e) {
+      throw new HoodieGlueSyncException("Fail to create " + tableId(databaseName, tableName), e);
+    }
+  }
+
+  @Override
+  public Map<String, String> getTableSchema(String tableName) {
+    try {
+      // GlueMetastoreClient returns partition keys separate from Columns, hence get both and merge to
+      // get the Schema of the table.
+      final long start = System.currentTimeMillis();
+      Table table = getTable(awsGlue, databaseName, tableName);
+      Map<String, String> partitionKeysMap =
+          table.getPartitionKeys().stream().collect(Collectors.toMap(Column::getName, f -> f.getType().toUpperCase()));
+
+      Map<String, String> columnsMap =
+          table.getStorageDescriptor().getColumns().stream().collect(Collectors.toMap(Column::getName, f -> f.getType().toUpperCase()));
+
+      Map<String, String> schema = new HashMap<>();
+      schema.putAll(columnsMap);
+      schema.putAll(partitionKeysMap);
+      final long end = System.currentTimeMillis();
+      LOG.info(String.format("Time taken to getTableSchema: %s ms", (end - start)));
+      return schema;
+    } catch (Exception e) {
+      throw new HoodieGlueSyncException("Fail to get schema for table " + tableId(databaseName, tableName), e);
+    }
+  }
+
+  @Override
+  public boolean doesTableExist(String tableName) {
+    return tableExists(tableName);
+  }
+
+  @Override
+  public boolean tableExists(String tableName) {
+    GetTableRequest request = new GetTableRequest()
+        .withDatabaseName(databaseName)
+        .withName(tableName);
+    try {
+      return Objects.nonNull(awsGlue.getTable(request).getTable());
+    } catch (EntityNotFoundException e) {
+      LOG.info("Table not found: " + tableId(databaseName, tableName), e);
+      return false;
+    } catch (Exception e) {
+      throw new HoodieGlueSyncException("Fail to get table: " + tableId(databaseName, tableName), e);
+    }
+  }
+
+  @Override
+  public boolean databaseExists(String databaseName) {
+    GetDatabaseRequest request = new GetDatabaseRequest();
+    request.setName(databaseName);
+    try {
+      return Objects.nonNull(awsGlue.getDatabase(request).getDatabase());
+    } catch (EntityNotFoundException e) {
+      LOG.info("Database not found: " + databaseName, e);
```

Comment:
error instead of info

Reply:
this method `databaseExists()` checks if it exists. so non-exist case should be expected.